core/arch/arm/kernel/user_ta.c

```diff
@@ -883,7 +883,8 @@ TEE_Result user_ta_set_prot(struct user_ta_ctx *utc, vaddr_t va, size_t len,
 	 * NULL) it cannot be made writeable as all mapped files are mapped
 	 * read-only.
 	 */
-	if (seg->file && (prot & (TEE_MATTR_UW | TEE_MATTR_PW)))
+	if ((seg->flags & VM_FLAG_READONLY) &&
```

Comment:
Since you're not checking `(seg->file)` anymore, the above comment sounds irrelevant.

Reply:
I'll fix